tests/test_zima.py

```diff
@@ -92,3 +94,41 @@ def test_zima_fuzzdays(tmp_path):
         lines = sys.stdout.getvalue()
     finally:
         sys.stdout = saved_stdout
+
+
+def test_simulate_corrnoise(tmp_path):
+    parfile = datadir / "B1855+09_NANOGrav_9yv1.gls.par"
+
+    m = get_model(parfile)
+
+    # Simulated TOAs won't have the correct flags for some of these to work.
+    m.remove_component("ScaleToaError")
+    m.remove_component("EcorrNoise")
+    m.remove_component("DispersionDMX")
+    m.remove_component("PhaseJump")
+    m.remove_component("FD")
+    m.PLANET_SHAPIRO.value = False
+
+    t = make_fake_toas_uniform(
+        m.START.value,
+        m.FINISH.value,
+        1000,
+        m,
+        add_noise=True,
+        add_correlated_noise=True,
+    )
+
+    # Check if the created TOAs can be whitened using
+    # the original timing model. This won't work if the
+    # noise is not realized correctly.
+    ftr = DownhillGLSFitter(t, m)
+    ftr.fit_toas()
+    rc = sum(ftr.resids.noise_resids.values())
+    r = ftr.resids.time_resids
+    rw = r - rc
+    sigma = ftr.resids.get_data_error()
+
+    # This should be independent and standard-normal distributed.
+    x = (rw / sigma).to_value("")
+    assert np.isclose(np.std(x), 1, atol=0.2)
+    assert np.isclose(np.mean(x), 0, atol=0.01)
```

Comment:
> Do you actually have any tests which look at the nature of the simulated noise? Or just that make sure there are simulated TOAs returned? Fitting the results may not be simple but maybe you can compare simulated TOAs with and without correlated noise to make sure they are different in the right way? Or some other test?

@dlakaplan Is this test sufficient? It is checking if the simulated TOAs can be whitened using the original model.

Reply:
I think that's reasonable.  I guess I don't know if the different noise realizations would play into it here: do you know that the phase of the red noise will be the same?

Reply:
Maybe I don't understand then.  If each realization has a different phase, how can you know that this will properly whiten the data?

Reply:
I carefully went through the `DownhillGLSFitter` code. My understanding is as follows:

When we call `DownhillGLSFitter.fit_toas()`, it effectively fits the correlated noise amplitudes (such as Fourier amplitudes for PLRedNoise) together with the timing model parameters. While the TimingModel object doesn't know about these amplitudes, the `DownhillGLSFitter` object knows about them via the `DownhillGLSFitter.current_state.xhat` property. This is a vector that contains the timing model parameter deviations along with the correlated noise amplitudes. The `noise_resids` property is computed by taking the inner product of these amplitudes with the correlated noise basis matrix. 

So, in effect, the `DownhillGLSFitter` object *does* know about the basis amplitudes, and hence the correlated noise phase.

Reply:
So in that case, are you actually sure that you are whitening the data with the input model?  Can you check that what the fitter finds is the same (or close to) what you injected?

Reply:
Here is a comparison between injected and recovered red noise.
![image](https://github.com/nanograv/PINT/assets/20718391/b99a65fc-076d-4a4a-ac43-9271f016b22b)


For posterity, here is the code for obtaining the above plot:
```
from pinttestdata import datadir
from pint.models import get_model
from pint.simulation import make_fake_toas_uniform
from pint.residuals import Residuals
from pint.fitter import DownhillGLSFitter
from astropy import units as u
from astropy import time
import numpy as np

parfile = datadir / "B1855+09_NANOGrav_9yv1.gls.par"
m = get_model(parfile)

m.remove_component("ScaleToaError")
m.remove_component("EcorrNoise")
m.remove_component("DispersionDMX")
m.remove_component("PhaseJump")
m.remove_component("FD")
m.PLANET_SHAPIRO.value = False

m.F0.frozen = True
m.F1.frozen = True

t = make_fake_toas_uniform(
	m.START.value,
	m.FINISH.value,
	1000,
	m,
	add_noise=True,
)

U = m.noise_model_designmatrix(t)
b = m.noise_model_basis_weight(t)
rn_inj = (U @ (b**0.5 * np.random.normal(size=len(b)))) << u.s
t.adjust_TOAs(time.TimeDelta(rn_inj))

res = Residuals(t, m)

ftr = DownhillGLSFitter(t, m)
ftr.fit_toas()

rn_fit = sum(ftr.resids.noise_resids.values())

plt.errorbar(t.get_mjds(), res.time_resids, t.get_errors(), label="Pre-fit Residuals", ls="")
plt.plot(t.get_mjds(), rn_inj, label="Injected red noise")
plt.plot(t.get_mjds(), rn_fit, label="Recovered red noise")
plt.legend()
plt.xlabel("MJD")
plt.ylabel("residuals (s)")
plt.show()
```

Reply:
I think the recovered signal is consistent with the injected signal. Note that I am freezing F0 and F1 in the above case so that fitting for them don't distort the recovered red noise. 

In practice, the red noise recovery will be distorted by the F0 & F1 fit like so:
![image](https://github.com/nanograv/PINT/assets/20718391/7456e819-992b-423b-8950-a3291627da92)

Reply:
OK, that looks good.  And can you also verify that if you simulate or fit using different noise models you don't get good residuals?

Reply:
Here is what happens when I set a wrong value for PLREDAMP before fitting.
![image](https://github.com/nanograv/PINT/assets/20718391/ccbea2b8-22c4-4d92-b2da-f0e4278edff9)

For comparison, the fit with correct values for the same dataset: 
![image](https://github.com/nanograv/PINT/assets/20718391/ddada6b1-5d67-44a9-8ba9-b7f1c9faa435)

Reply:
OK, I think this all makes sense.